hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/ExecuteOperationCommand.java

```diff
@@ -19,68 +19,61 @@
 import java.util.Collection;
 import java.util.Map;
 
-import org.hawkular.agent.monitor.extension.MonitorServiceConfiguration;
 import org.hawkular.agent.monitor.inventory.ID;
-import org.hawkular.agent.monitor.inventory.InventoryIdUtil;
-import org.hawkular.agent.monitor.inventory.InventoryIdUtil.ResourceIdParts;
 import org.hawkular.agent.monitor.inventory.ManagedServer;
-import org.hawkular.agent.monitor.inventory.Name;
 import org.hawkular.agent.monitor.inventory.ResourceManager;
 import org.hawkular.agent.monitor.inventory.dmr.DMRInventoryManager;
 import org.hawkular.agent.monitor.inventory.dmr.DMROperation;
 import org.hawkular.agent.monitor.inventory.dmr.DMRResource;
-import org.hawkular.agent.monitor.inventory.dmr.LocalDMRManagedServer;
-import org.hawkular.agent.monitor.inventory.dmr.RemoteDMRManagedServer;
 import org.hawkular.agent.monitor.log.AgentLoggers;
 import org.hawkular.agent.monitor.log.MsgLogger;
 import org.hawkular.bus.common.BasicMessageWithExtraData;
 import org.hawkular.cmdgw.api.ExecuteOperationRequest;
 import org.hawkular.cmdgw.api.ExecuteOperationResponse;
-import org.hawkular.cmdgw.api.MessageUtils;
-import org.hawkular.cmdgw.api.ResponseStatus;
+import org.hawkular.dmr.api.OperationBuilder;
 import org.hawkular.dmrclient.Address;
-import org.hawkular.dmrclient.CoreJBossASClient;
-import org.hawkular.dmrclient.JBossASClient;
 import org.hawkular.inventory.api.model.CanonicalPath;
 import org.jboss.as.controller.client.ModelControllerClient;
-import org.jboss.dmr.ModelNode;
 
 /**
  * Execute an operation on a resource.
  */
-public class ExecuteOperationCommand implements Command<ExecuteOperationRequest, ExecuteOperationResponse> {
+public class ExecuteOperationCommand extends
+        AbstractResourcePathCommand<ExecuteOperationRequest, ExecuteOperationResponse> {
     private static final MsgLogger log = AgentLoggers.getLogger(ExecuteOperationCommand.class);
     public static final Class<ExecuteOperationRequest> REQUEST_CLASS = ExecuteOperationRequest.class;
 
+    public ExecuteOperationCommand() {
+        super("Execute Operation", "DMR Node");
+    }
+
+    /** @see org.hawkular.agent.monitor.cmd.AbstractResourcePathCommand#createResponse() */
     @Override
-    public BasicMessageWithExtraData<ExecuteOperationResponse> execute(
-            BasicMessageWithExtraData<ExecuteOperationRequest> envelope, CommandContext context) throws Exception {
-        ExecuteOperationRequest request = envelope.getBasicMessage();
-        log.infof("Received request to execute operation [%s] on resource [%s]", request.getOperationName(),
```

Comment:
This (an example on running the tests on my computer):
```
Received request to perform [Execute Operation] on a [DMR Node] given by inventory path [/t;28026b36-8fe4-4332-84c8-524e173a68bf/e;test/f;earthquake/r;earthquake~Local~~/r;earthquake~Local~%2Fdeployment=hawkular-helloworld-war.war]
```
is written here: https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/AbstractResourcePathCommand.java#L194-L197

Is that enough? or should i return the log to the ExecuteOperationCommand?
```java
log.infof("Received request to execute operation [%s] on resource [%s]", request.getOperationName(),
```
Or maybe we could make these attributes not final to change the [Execute Operation] and [DMR Node] on runtime.
https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/AbstractResourcePathCommand.java#L51-L62